core/geometry/src/polyface/PolyfaceQuery.ts

```diff
@@ -956,35 +1028,75 @@ export class PolyfaceQuery {
+    let result: CurvePrimitive[] = [];
+    // setup default options:
+    if (options === undefined)
+      options = SweepLineStringToFacetsOptions.create(
+        Vector3d.unitZ(),
+        Angle.createRadians(Geometry.smallAngleRadians),   // tight geometry tolerance for vertical side facets
+        true,   // assemble chains
+        true, true, true); // accept all outputs
+    let chainContext: ChainMergeContext | undefined;
+    if (options.assembleChains)
+      chainContext = ChainMergeContext.create();
+    const context = ClipSweptLineStringContext.create(linestringPoints, options.vectorToEye);
+    if (context) {
+      const visitor = polyface.createVisitor(0);
+      for (visitor.reset(); visitor.moveToNextFacet();) {
+        if (options?.collectFromThisFacetNormal(PolygonOps.areaNormalGo(visitor.point))) {
+          context.processPolygon(visitor.point.getArray(),
+            (pointA: Point3d, pointB: Point3d) => {
+              if (chainContext !== undefined)
+                chainContext.addSegment(pointA, pointB);
+              else
+                result.push(LineSegment3d.create(pointA, pointB));
+            });
+        }
+      }
+      if (chainContext !== undefined) {
+        chainContext.clusterAndMergeVerticesXYZ();
+        result = chainContext.collectMaximalChains();
+      }
+    }
+    return result;
+  }
+  /** Find segments (within the linestring) which project to facets.
+    * * Return collected line segments
+    * * this calls  [PolyfaceQuery.sweepLineStringToFacets] with options created by
+    *   `const options = SweepLineStringToFacetsOptions.create(Vector3d.unitZ(), Angle.createSmallAngle(),false, true, true, true);`
+    * @deprecated in 4.x. Use [PolyfaceQuery.sweepLineStringToFacets] to get further options.
+    */
   public static sweepLinestringToFacetsXYReturnLines(linestringPoints: GrowableXYZArray, polyface: Polyface): LineSegment3d[] {
-    const drapeGeometry: LineSegment3d[] = [];
-    this.announceSweepLinestringToConvexPolyfaceXY(linestringPoints, polyface,
-      (_linestring: GrowableXYZArray, _segmentIndex: number,
-        _polyface: Polyface, _facetIndex: number, points: Point3d[], indexA: number, indexB: number) => {
-        drapeGeometry.push(LineSegment3d.create(points[indexA], points[indexB]));
-      });
-    return drapeGeometry;
+    const options = SweepLineStringToFacetsOptions.create(Vector3d.unitZ(), Angle.createSmallAngle(),
+      false, true, true, true);
+    const result = PolyfaceQuery.sweepLineStringToFacets(linestringPoints, polyface, options);
+    return result as LineSegment3d[];
   }
 
   /** Find segments (within the linestring) which project to facets.
    * * Return chains.
-   * * Facets are ASSUMED to be convex and planar, and not overlap in the z direction.
+   * * this calls  [PolyfaceQuery.sweepLineStringToFacets] with options created by
```

Comment:
```suggestion
   * * this calls  [[PolyfaceQuery.sweepLineStringToFacets]] with options created by
```

`[[ApiName]]` to link to an API in the same package.
`[ApiName]($package-alias)` to link to an API in a different package.